tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/AzurePipelinesTool/AzurePipelinesTool.cs

```diff
@@ -145,15 +144,24 @@ public async Task<Build> GetPipelineRun(string? project, int buildId)
     [McpServerTool, Description("Gets failures from tasks (non-test failures) in a pipeline run")]
     public async Task<List<TimelineRecord>?> GetPipelineTaskFailures(string project, int buildId)
     {
-        logger.LogDebug("Getting pipeline task failures for {project} {buildId}", project, buildId);
-        var timeline = await buildClient.GetBuildTimelineAsync(project, buildId);
-        var failedNonTests = timeline.Records.Where(
-                                r => r.Result == TaskResult.Failed
-                                && r.RecordType == "Task"
-                                && !IsTestStep(r.Name))
-                            .ToList();
-        logger.LogDebug("Found {count} failed tasks", failedNonTests.Count);
-        return failedNonTests;
+        try
+        {
+            logger.LogDebug("Getting pipeline task failures for {project} {buildId}", project, buildId);
+            var timeline = await buildClient.GetBuildTimelineAsync(project, buildId);
+            var failedNonTests = timeline.Records.Where(
+                                    r => r.Result == TaskResult.Failed
+                                    && r.RecordType == "Task"
+                                    && !IsTestStep(r.Name))
+                                .ToList();
+            logger.LogDebug("Found {count} failed tasks", failedNonTests.Count);
+            return failedNonTests;
+        }
+        catch (Exception ex)
+        {
+            logger.LogError("Failed to get pipeline task failures {buildId}: {exception}", buildId, ex.Message);
+            SetFailure();
+            return null;
```

Comment:
I think I should make `Response<T>` with a `T Value` property, that way we can return errors to the MCP server:

```
return new Response<List<TimelineRecord>>()
{
    ResponseError = $"Failed to get pipeline task failures {buildId}: {ex.Message}"
}
```